src/simple_minreq.rs

```diff
@@ -0,0 +1,271 @@
+            io::sink().write(buf)
+        }
+        fn flush(&mut self) -> io::Result<()> {
+            Ok(())
+        }
+    }
+
+    impl TcpStream {
+        pub fn connect_timeout(_: &SocketAddr, _: Duration) -> io::Result<Self> {
+            Ok(TcpStream)
+        }
+        pub fn set_read_timeout(&self, _: Option<Duration>) -> io::Result<()> {
+            Ok(())
+        }
+        pub fn set_write_timeout(&self, _: Option<Duration>) -> io::Result<()> {
+            Ok(())
+        }
+    }
+}
+
+/// The default TCP port to use for connections.
+/// Set to 8332, the default RPC port for bitcoind.
+pub const DEFAULT_PORT: u16 = 8332;
+
+/// Simple HTTP transport that implements the necessary subset of HTTP for
+/// running a bitcoind RPC client.
+#[derive(Clone, Debug)]
+pub struct SimpleHttpTransport {
+    url: String,
+    timeout: Duration,
+    /// The value of the `Authorization` HTTP header, i.e., a base64 encoding of 'user:password'.
+    basic_auth: Option<String>,
+}
+
+impl Default for SimpleHttpTransport {
+    fn default() -> Self {
+        SimpleHttpTransport {
+            url: format!("localhost:{}", DEFAULT_PORT),
+            #[cfg(fuzzing)]
+            timeout: Duration::from_secs(1), // Minreq only accepts timeout in seconds.
+            #[cfg(not(fuzzing))]
+            timeout: Duration::from_secs(15),
+            basic_auth: None,
+        }
+    }
+}
+
+impl SimpleHttpTransport {
+    /// Constructs a new [`SimpleHttpTransport`] with default parameters.
+    pub fn new() -> Self {
+        SimpleHttpTransport::default()
+    }
+
+    /// Returns a builder for [`SimpleHttpTransport`].
+    pub fn builder() -> Builder {
+        Builder::new()
+    }
+
+    fn request<R>(&self, req: impl serde::Serialize) -> Result<R, Error>
+    where
+        R: for<'a> serde::de::Deserialize<'a>,
+    {
+        let req = match &self.basic_auth {
+            Some(auth) => {
+                minreq::Request::new(minreq::Method::Post, &self.url)
+                    .with_timeout(self.timeout.as_secs())
+                    .with_header("Authorization", auth)
+                    .with_json(&req)?
+            }
+            None => {
+                minreq::Request::new(minreq::Method::Post, &self.url)
+                    .with_timeout(self.timeout.as_secs())
+                    .with_json(&req)?
+            }
+        };
+
+        let resp = req.send()?;
+        let json = resp.json()?;
+        Ok(json)
+    }
+}
+
+/// Error that can happen when sending requests.
+#[derive(Debug)]
+pub enum Error {
+    /// JSON parsing error.
+    Json(serde_json::Error),
+    /// Minreq error.
+    Minreq(minreq::Error),
+}
+
+impl fmt::Display for Error {
+    fn fmt(&self, f: &mut fmt::Formatter) -> Result<(), fmt::Error> {
+        match *self {
+            Error::Json(ref e) => write!(f, "JSON error: {}", e),
+            Error::Minreq(ref e) => write!(f, "MINREQ error: {}", e),
```

Comment:
In d20ba2eafcaa3cd5245f5a548cbcbb6d14c13432:

Arguably the prefixes shouldn't be here at all, but if so, they should be `JSON:` and `minreq:`. There's no need for the word `error` and `MINREQ` should not be capitalized.

Reply:
woops, changed to:
```rust
            Error::Json(ref e) => write!(f, "parsing JSON failed: {}", e),
            Error::Minreq(ref e) => write!(f, "minreq: {}", e),
```

Reply:
Sounds good!